# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/IMEData.cpp
# Commit: 4eb5dc745ee8
# Full Hash: 4eb5dc745ee846a5172d999c5af98fd3a4007e5f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-02-08 17:32:27
# Regressor Bug: 1746104
# File Overlap Count: 1
# Description:
#   Bug 1746104 - part 6-0: Clean up logging code in `ContentCache.cpp` r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137426
# ==============================================================================

diff -r 0d81c2841c4d -r 4eb5dc745ee8 widget/IMEData.cpp
--- a/widget/IMEData.cpp	Mon Feb 07 22:33:38 2022 +0000
+++ b/widget/IMEData.cpp	Mon Feb 07 22:33:39 2022 +0000
@@ -27,7 +27,7 @@
       aString.Length() <= aMaxLength ? 0 : aMaxLength / 2;
   for (uint32_t i = 0; i < aString.Length(); i++) {
     if (i > 0) {
-      Append(" ");
+      AppendLiteral(" ");
     }
     char32_t ch = aString.CharAt(i);
     if (NS_IS_HIGH_SURROGATE(ch) && i + 1 < aString.Length() &&
@@ -36,7 +36,7 @@
     }
     Append(PrintCharData(ch));
     if (i + 1 == kFirstHalf) {
-      Append(" ...");
+      AppendLiteral(" ...");
       i = aString.Length() - kSecondHalf - 1;
       if (NS_IS_LOW_SURROGATE(aString.CharAt(i)) &&
           NS_IS_HIGH_SURROGATE(aString.CharAt(i - 1))) {
@@ -50,7 +50,9 @@
       i++;
     }
   }
-  Append("\"");
+  AppendLiteral("\" (Length()=");
+  AppendInt(static_cast<uint32_t>(aString.Length()));
+  AppendLiteral(")");
 }
 
 // static
